### PR TITLE
SC 4142 status updater job

### PR DESCRIPTION
**Pull request by kayline**
## Summary

- *This work is behind a feature toggle (flipper): YES: decision_review_track_4142_submissions*
- The job will continue to run on its regular schedule, but this new functionality won't run until the Flipper is enabled
- polls and updates status of 4142 forms submitted during Supplemental Claim to detect silent failures
- logs metrics
- Decision Reviews, yes

## Related issue(s)

department-of-veterans-affairs/va.gov-team/issues/95084

## Testing done

- [X] *New code is covered by unit tests*
- We were never checking on the status of submitted 4142s
- Run the job in staging, see status change for relevant records
- Tests for job in both cases. When disabled the job is a no-op
  - Enable for a percentage of users, verify it can run regularly without error, enable for all

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
No user-facing changes. Expands the SavedClaimScStatusUpdaterJob to check status of 4142s. This only applies for 4142s submitted as part of submitting a Supplemental Claim. Doesn't include a chron entry, will test job in production before adding a scheduled entry.

## Acceptance criteria

- [X]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [X]  No error nor warning in the console.
- [X]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [X]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable) - need to build monitor once this goes live
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

File: app/sidekiq/decision_review/saved_claim_sc_status_updater_job.rb [hunk 1]
```diff
@@ -85,6 +84,24 @@ def get_evidence_uploads_statuses(submitted_appeal_uuid)
       result
     end
 
+    def get_and_update_secondary_form_statuses(submitted_appeal_uuid)
+      all_complete = true
+      return all_complete unless Flipper.enabled?(:decision_review_track_4142_submissions)
+
+      secondary_forms = AppealSubmission.find_by(submitted_appeal_uuid:)&.secondary_appeal_forms
+      secondary_forms = secondary_forms&.filter { |form| form.delete_date.nil? } || []
```
Comment (shaunburdick) [inline app/sidekiq/decision_review/saved_claim_sc_status_updater_job.rb:92]: Not a huge fan of variable reuse here. Can we use a different variable or add the `.filter {}` to the end of the chain on the line above?

**Review by shaunburdick — APPROVED** [review]
LGTM, I left a somewhat pedantic comment. Up to you if you want to address or not.

File: app/sidekiq/decision_review/saved_claim_sc_status_updater_job.rb (hunk 1 above)
Comment (kayline) [reply app/sidekiq/decision_review/saved_claim_sc_status_updater_job.rb:92]: I want to turn this into a scope on AppealSubmission so the whole query can happen in SQL, which should remove the need for this mess

**Comment by rmtolmach** [comment]
You hit a bug where specs are failing with the error: `Error: The action 'Run Specs' has timed out after 20 minutes.` So I'm rerunning them for you.

**Review by rmtolmach — APPROVED** [review]

**Pull request merged**